operator/src/main/java/org/keycloak/operator/crds/v2alpha1/deployment/KeycloakSpec.java

```diff
@@ -44,6 +44,9 @@ public class KeycloakSpec {
     @JsonPropertyDescription("Custom Keycloak image to be used.")
     private String image;
 
+    @JsonPropertyDescription("If the custom Keycloak image is optimized, defaults to true.")
+    private Boolean optimized;
```

Comment:
I'm not sure that is much clearer as you still have to be aware of the image property to know the full meaning - because you would rightfully know there's always an image reference on the StatefulSet.

You'd have to do something like customImageOptimized for the property name alone to be sufficiently descriptive - and then it's much too verbose / and uses a different convention that the image property.  I'd be fine with leaving it as just optimized.  

It's not handled here, but there's is also a chance that some users may want to change the base image referenced by the operator to be an optimized one, in which case this property could override the default non-optimized behavior to optimized as well.

Reply:
My reasoning was to align with other similarly named fields in pod templates, e.g. `image` and `imageOptimized`. Only `optimized` might make the impression that it's something related to KC instance rather than the image itself.

> It's not handled here, but there's is also a chance that some users may want to change the base image referenced by the operator to be an optimized one, in which case this property could override the default non-optimized behavior to optimized as well.

Yeah, the `optimized` flag might work for the default image as well. If not explicitly specified, we'd default to the current behaviour (`false` for default image, `true` for custom). We'd document this behaviour.

Reply:
To support this, there's an additional config value as well.